opentelemetry-auto-instrumentation/src/opentelemetry/auto_instrumentation/bootstrap.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3
+
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import pkgutil
+import subprocess
+import sys
+from logging import getLogger
+
+logger = getLogger(__file__)
+
+_OUTPUT_INSTALL = "install"
+_OUTPUT_REQUIREMENTS = "requirements"
+
+
+# target library to desired instrumentor path/versioned package name
+instrumentations = {
+    "dbapi": "opentelemetry-ext-dbapi>=0.6b0",
+    "flask": "opentelemetry-ext-flask>=0.6b0",
+    "grpc": "opentelemetry-ext-grpc>=0.6b0",
+    "requests": "opentelemetry-ext-http-requests>=0.6b0",
+    "mysql": "opentelemetry-ext-mysql>=0.6b0",
+    "psycopg2": "opentelemetry-ext-psycopg2>=0.6b0",
+    "pymongo": "opentelemetry-ext-pymongo>=0.6b0",
+    "pymysql": "opentelemetry-ext-pymysql",
+    "redis": "opentelemetry-ext-redis",
+    "sqlalchemy": "opentelemetry-ext-sqlalchemy",
+    "wsgi": "opentelemetry-ext-wsgi>=0.6b0",
+}
```

Comment:
We should not have hardcoded references to existing instrumentations. This list should be compiled from the available packages that implement the `opentelemetry-instrumentor` entry point.

Reply:
is there a way to get this information at runtime or do you want to make it a build step and just automate generation of this list?

Reply:
Maybe [this](https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-auto-instrumentation/src/opentelemetry/auto_instrumentation/sitecustomize.py#L22) can help. Let me know if it does, please :slightly_smiling_face:

Reply:
Not sure if it does. This can help me get a list of installed instrumentations at runtime right?

This script is supposed to be used to install the required instrumentation packages. If we were to automate this, I think the way to do it would be to create a dev script that walks over the ext packages, figures out which ones are auto-instrumentation packages, then figures out which libraries they instrument and then generate a .py file for bootstrap.py to import and use. Even if we do that, in my experience working with a similar command for SignalFx's Python instrumentation, we might need to pin specific versions of different instrumentation in different versions of the bootstrap script so it might not always be feasible to auto-generate such a list without adding some "hard-coded" annotations to instrumentations or bootstrap script.

Reply:
This is a bit of a tricky problem. @owais can you file an issue to see if we can track a more automated way to do this, or just maybe include this in documentation for auto-instrumentation?

There's merit in hardcoding as well: we don't add magic auto-instrumentation until we've tested things sufficiently.

Reply:
https://github.com/open-telemetry/opentelemetry-python/issues/668